pennylane/templates/subroutines/adder.py

```diff
@@ -80,17 +80,22 @@ def __init__(
 
         x_wires = qml.wires.Wires(x_wires)
 
+        num_works_wires = 0 if work_wires is None else len(work_wires)
+
         if mod is None:
             mod = 2 ** len(x_wires)
-        elif work_wires is None and mod != 2 ** len(x_wires):
-            raise ValueError(f"If mod is not 2^{len(x_wires)}, two work wires should be provided.")
+        elif mod != 2 ** len(x_wires) and num_works_wires != 2:
```

Comment:
Without this, `work_wires` of other lengths fail when executing the circuit with an error about `PauliX` receiving the wrong number of wires

Reply:
I'm not confident I checked for every single one, but some of them crash with too many work wires. 

In this case, it passes `work_wires[1:]` to `PhaseAdder`, so `PhaseAdder` gets too many wires if we let more than 2 through. If we also let too many `work_wires` through `PhaseAdder`, it tries to pass the `work_wire` to `PauliX` in the decomposition and crashes because PauliX can only take one wire.

Reply:
True, forget about it :)